enarx-keep-sev-shim/src/addr.rs

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Some basic address operations
+
+// FIXME: re-export {PhysAddr, VirtAddr} for now, until we have arch independent types.
+use x86_64::{PhysAddr, VirtAddr};
+
+/// The offset of shim virtual address space to the physical address
+///
+/// physical address + `SHIM_VIRT_OFFSET` = shim virtual address
+///
+/// FIXME: change to dynamic offset with ASLR
+const SHIM_VIRT_OFFSET: u64 = 0xFFFF_FF80_0000_0000;
+
+/// Address in the shim virtual address space
+#[derive(Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
+#[repr(transparent)]
+pub struct ShimVirtAddr(VirtAddr);
+
+impl ShimVirtAddr {
+    /// Creates a shim virtual address from the given pointer
+    ///
+    /// # Panics
+    ///
+    /// Panics, if the address is not in the shim virtual address space.
+    #[inline(always)]
+    pub fn from_ptr<T>(v: *const T) -> ShimVirtAddr {
+        let virt = VirtAddr::from_ptr(v);
+        assert!(virt.as_u64() >= SHIM_VIRT_OFFSET);
```

Comment:
panic

Reply:
Does it also panic in a release build?

Reply:
yes

Reply:
This is like a NULL pointer... it just should not happen

Reply:
Ack, I was just curious if this turns into a warning during release builds. Otherwise, to Nathaniel's question, if the shim panicks it will attempt to proxy an exit call and if the hypervisor ignores this or maliciously re-enters, the shim halts.